src/debug.rs

```diff
@@ -0,0 +1,352 @@
+                }
+            };
+
+            let execution_mode = TxExecutionMode::EthCall;
+            let storage = StorageView::new(&inner.fork_storage).to_rc_ptr();
+
+            let bootloader_code = inner.system_contracts.contacts_for_l2_call();
+
+            // init vm
+            let (l1_batch_env, _block_context) = inner.create_l1_batch_env(storage.clone());
+            let system_env = inner.create_system_env(bootloader_code.clone(), execution_mode);
+            let mut vm = Vm::new(l1_batch_env, system_env, storage, HistoryDisabled);
+
+            // We must inject *some* signature (otherwise bootloader code fails to generate hash).
+            if l2_tx.common_data.signature.is_empty() {
+                l2_tx.common_data.signature =
+                    PackedEthSignature::default().serialize_packed().into();
+            }
+
+            let tx: Transaction = l2_tx.clone().into();
+            vm.push_transaction(tx);
+
+            let call_tracer_result = Arc::new(OnceCell::default());
+            let tracer = CallTracer::new(call_tracer_result.clone(), HistoryDisabled);
+            let tx_result = vm.inspect(vec![Box::new(tracer)], vm::VmExecutionMode::OneTx);
+
+            let call_traces = if only_top {
+                vec![]
+            } else {
+                Arc::try_unwrap(call_tracer_result)
+                    .unwrap()
+                    .take()
+                    .unwrap_or_default()
+            };
+
+            let calltype = if l2_tx.recipient_account() == CONTRACT_DEPLOYER_ADDRESS {
+                DebugCallType::Create
+            } else {
+                DebugCallType::Call
+            };
+
+            let result = match &tx_result.result {
+                ExecutionResult::Success { output } => DebugCall {
+                    gas_used: tx_result.statistics.gas_used.into(),
+                    output: output.clone().into(),
+                    r#type: calltype,
+                    from: l2_tx.initiator_account(),
+                    to: l2_tx.recipient_account(),
+                    gas: l2_tx.common_data.fee.gas_limit,
+                    value: l2_tx.execute.value,
+                    input: l2_tx.execute.calldata().into(),
+                    error: None,
+                    revert_reason: None,
+                    calls: call_traces.into_iter().map(Into::into).collect(),
+                },
+                ExecutionResult::Revert { output } => DebugCall {
+                    gas_used: tx_result.statistics.gas_used.into(),
+                    output: output.encoded_data().into(),
+                    r#type: calltype,
+                    from: l2_tx.initiator_account(),
+                    to: l2_tx.recipient_account(),
+                    gas: l2_tx.common_data.fee.gas_limit,
+                    value: l2_tx.execute.value,
+                    input: l2_tx.execute.calldata().into(),
+                    error: Some(output.to_string()),
```

Comment:
In `zksync_core` we expect `error` to be `None`: https://github.com/matter-labs/zksync-era/blob/main/core/lib/types/src/vm_trace.rs#L119

Should we update or keep the same?
```suggestion
                    error: None,
```